pandas-stubs/core/frame.pyi

```diff
@@ -620,7 +625,10 @@ class DataFrame(NDFrame, OpsMixin):
     def query(
         self, expr: _str, *, inplace: Literal[False] = ..., **kwargs
     ) -> DataFrame: ...
-    def eval(self, expr: _str, *, inplace: _bool = ..., **kwargs): ...
+    @overload
+    def eval(self, expr: _str, *, inplace: Literal[False] = ..., **kwargs): ...
+    @overload
+    def eval(self, expr: _str, *, inplace: Literal[True] = ..., **kwargs) -> None: ...
```

Comment:
swap these 2 overloads, and with `inplace: Literal[True]`, remove the ellipses.

And for `Literal[False] = ...`, we should return `Scalar | np.ndarray | pd.DataFrame | pd.Series`

Reply:
I was not totally sure about the returns of the case for `False` but thanks for adding the details, it is implemented.
For the first point about the order, do you mind clarifying how it changes the results to have them swapped?

Reply:
The matching of overloads is not defined by the typing spec.  There is an effort to document that better.  The general rule is that any call to a function/method will pick the first overload that matches.

So the convention to use is to put the narrower matches before wider matches.  So with `inplace: Literal[True]` being first, it says match `inplace = True`, and if that matches, don't try anything else.  When the second overload is `inplace: Literal[False] = ...`, it is saying match this overload if the value is `False` or unspecified.  But this match is in essence wider than just `inplace: Literal[True]` because a user doesn't have to specify the value.  It would also be valid to write `inplace: bool = ...` as the last overload. 

We may not be consistent with this practice throughout `pandas-stubs`, but with reviews, I'm trying to get there.